feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/home/HomeFragment.java

```diff
@@ -1,42 +1,256 @@
+                                getContext());
+                        // set item background
+                        editItem.setBackground(new ColorDrawable(Color.rgb(0xC9, 0xC9,
+                                0xCE)));
+                        // set item width
+                        editItem.setWidth(170);
+                        // set item title
+                        editItem.setTitle("Edit");
+                        // set item title fontsize
+                        editItem.setTitleSize(18);
+                        // set item title font color
+                        editItem.setTitleColor(Color.WHITE);
+                        // add to menu
+                        menu.addMenuItem(editItem);
+
+                        // delete
+                        SwipeMenuItem deleteItem = new SwipeMenuItem(
+                                getContext());
+                        // set item background
+                        deleteItem.setBackground(new ColorDrawable(Color.rgb(0xF9,
+                                0x3F, 0x25)));
+                        // set item width
+                        deleteItem.setWidth(170);
+                        // set a icon
+                        deleteItem.setIcon(R.drawable.ic_delete);
+                        // add to menu
+                        menu.addMenuItem(deleteItem);
+                    }
+                };
+                SwipeMenuListView.setMenuCreator(creator);
+                SwipeMenuListView.setOnMenuItemClickListener(new com.baoyz.swipemenulistview.SwipeMenuListView.OnMenuItemClickListener() {
+                    @Override
+                    public boolean onMenuItemClick(int position, SwipeMenu menu, int index) {
+                        switch (index) {
+                            case 0:
+                                // view
+                                viewEmotion(view, position);
+                                break;
+                            case 1:
+                                // view
+                                editEmotion(view, position);
+                                break;
+
+                            case 2:
+                                // Delete
+                                removeEmotion(view, position);
+                                break;
+                        }
+                        // false : close the menu; true : not close the menu
+                        return false;
+                    }
+                });
+            }
+        }, new VoidCallback() {
+            @Override
+            public void onCallback() { }
+        });
+    }
+
+    public  void removeEmotion(@NonNull View view, int index){
+        List<MoodEvent> moodHistoryTempTemp = currentUser.getMoodHistory();
+        SwipeMenuListView = view.findViewById(R.id.listView);
+        myEmotionList.remove(index);
+        moodHistoryTempTemp.remove(index);
+        UserDAO userAdo = new UserDAO();

```

Comment:
We should set the UserDao to a method in the activity, so we don't need to instantiate it every time we call a funtion